src/openrct2-ui/windows/Ride.cpp

```diff
@@ -1335,15 +1335,15 @@ static rct_window* WindowRideOpenStation(Ride* ride, StationIndex stationIndex)
     WindowInitScrollWidgets(w);
 
     // View
-    for (int32_t i = stationIndex; i >= 0; i--)
+    for (StationIndex::UnderlyingType i = stationIndex.ToUnderlying(); i >= 0; i--)
     {
         if (ride->stations[i].Start.IsNull())
         {
-            stationIndex--;
+            stationIndex = StationIndex::FromUnderlying(stationIndex.ToUnderlying() - 1);
         }
     }
```

Comment:
There are a couple more loops like this, that just loop over the stations in reverse until a valid station is found. Could be moved to a member function of the ride.